lib/pluck_map/attribute.rb

```diff
@@ -1,15 +1,20 @@
 module PluckMap
   class Attribute
-    attr_reader :id, :model, :selects, :name, :value, :block
+    attr_reader :id, :model, :selects, :name, :alias, :value, :block
     attr_accessor :indexes
 
     def initialize(id, model, options={})
       @id = id
       @model = model
       @selects = Array(options.fetch(:select, id))
       @name = options.fetch(:as, id)
+      @alias = name.to_s.tr("_", " ")
       @block = options[:map]
 
+      if selects.length == 1 && !selects.first.is_a?(Symbol)
+        @selects = [ Arel.sql("#{selects.first} AS \"#{@alias}\"") ]
```

Comment:
As of [4d030c7](https://github.com/boblail/pluck_map/commit/4d030c709364ba5ef284f183d724ac360f8347b4#diff-67dace4d3a79b7b3d1dd1d4dd014bad0), `selects` will never contain a `String`; all raw SQL values will already be wrapped in `Arel.sql`.

Instead of this

```ruby
Arel.sql("#{select} AS \"#{name}\"") 
```

do this:

```ruby
select.as(name)
```

`Arel::Nodes::SqlLiteral` includes `Arel::AliasPredication`, so we can use `as` to set one up. That way our code isn't responsible for knowing
  1. whether the current database supports `AS`
  2. whether the current database quotes values with `"` 
  3. whether we've escaped `name` (or `@alias`) appropriately

Reply:
Neat! It's much more elegant to be assured of getting an `Arel::Nodes::SqlLiteral` 👍